### PR TITLE
Promisified the internals of the `validateFunc` option …

### PR DESCRIPTION
**Pull request by jonathanmarvens**

### CODE REVIEW

File: lib/startup.js [hunk 1]
```diff
@@ -59,16 +59,14 @@ function makeServer(config) {
         cookie: process.env.SESSION_COOKIE,
         clearInvalid: true,
         validateFunc: function(session, cb) {
-          cache.get(session.sid, function(err, item, cached) {
-
-            if (err) {
-              return cb(err, false);
-            }
+          P.promisify(cache.get, cache)(session.sid).catch(function(err) {
```
Comment (aredridel) [inline lib/startup.js]: Hm. What if `cache.get` returned a promise to start? that'd make this a lot tidier.

Comment (aredridel) [inline lib/startup.js]: Unless of course we didn't write that.

Comment (jonathanmarvens) [inline lib/startup.js]: @aredridel Yes, it’s returned by [hapi.js](http://hapijs.coms.com "hapi.js")’s [`Server.prototype.cache(...)`](http://hapijs.com/api/9.3.1#servercacheoptions "9.3.1 API Reference").

**Comment by aredridel** [comment]
Given the above, looks good to me!

**Pull request merged**